API.md

```diff
@@ -0,0 +1,200 @@
+ * **`results`**: A JSON Array of record objects. Due to the use of the `_fields` option in this query, there are only three fields in each record - the three fields specified in the `_fields` parameter. When the `_fields` parameter is omitted, the full record is provided.
+
+### Error Example
+
+Let's change the query so as to generate an error when it's executed:
+
+```
+https://ccapi-production.18f.gov/v1/schools.json?school.degrees_awarded.predominant=frog&_fields=id,school.name,wombat
+```
+
+This is the JSON document returned:
+
+```json
+{
+  "errors": [
+    {
+      "error": "field_not_found",
+      "input": "wombat",
+      "message": "The input field 'wombat' (in the fields parameter) is not a field in this dataset."
+    },
+    {
+      "error": "parameter_type_error",
+      "parameter": "school.degrees_awarded.predominant",
+      "input": "frog",
+      "expected_type": "integer",
+      "input_type": "string",
+      "message": "The parameter 'school.degrees_awarded.predominant' expects a value of type integer, but received 'frog' which is a value of type string."
+    }
+  ]
+}
+```
+
+When failing to execute a query, Open Data Maker will attempt to return a JSON error document that explains the problem. Points to note:
+
+ * The HTTP response status will be in the 400 or 500 range, indicating a problem.
+ * The JSON document contains a single top-level element, `errors`, containing a JSON Array of error objects. Instead of simply returning the first error encountered, Open Data Maker attempts to list all the problems that it detected.
+ * Error objects always contain these two elements:
+   * `error`: a symbolic error code for the specific error that occurred.
+   * `message`: an English description of the error.
+ * Error objects may also contain these fields:
+   * `input`: the provided input which triggered the error.
+   * `parameter`: the parameter in which the `input` was supplied.
+   * `input_type` & expected_type`: In the case of a type mismatch, these fields list the data types that were provided and expected.
+
+## Field Parameters
+
+Parameter names _without_ an underscore prefix are assumed to be field names in the dataset. Supplying a value to a field parameter acts as a query filter, and only returns records where the given field exactly matches the given value.
+
+For example: Use the parameter `school.region_id=6` to only fetch records with a `school.region_id` value of `6`.
+
+### Word and substring matches on name fields
+
+Certain text fields in the dataset - those with the `name` or `autcomplete` data type - allow querying with a list of words. To search for a given word or string of words in those fields, simply provide a list of space-separated words. This will return all records where the given field contains the given words as part of a string. **Note that all given words have to be at least three characters long.**
```

Comment:
the below description is correct for 'autocomplete' type.  Let's drop 'name' from docs for now and add back later (or remove the type)